pkg/common-controller/snapshot_controller.go

```diff
@@ -236,6 +237,22 @@ func (ctrl *csiSnapshotCommonController) syncSnapshot(snapshot *crdv1.VolumeSnap
 // 2. Call checkandRemoveSnapshotFinalizersAndCheckandDeleteContent() with information obtained from step 1. This function name is very long but the name suggests what it does. It determines whether to remove finalizers on snapshot and whether to delete content.
 func (ctrl *csiSnapshotCommonController) processSnapshotWithDeletionTimestamp(snapshot *crdv1.VolumeSnapshot) error {
 	klog.V(5).Infof("processSnapshotWithDeletionTimestamp VolumeSnapshot[%s]: %s", utils.SnapshotKey(snapshot), utils.GetSnapshotStatusForLogging(snapshot))
+	driverName, err := ctrl.getSnapshotDriverName(snapshot)
+	if err != nil {
+		klog.Errorf("failed to getSnapshotDriverName while recording metrics for snapshot %q: %v", utils.SnapshotKey(snapshot), err)
+	}
+
+	// Processing delete, start operation metric
+	deleteOperation := metrics.NewOperation(metrics.DeleteSnapshotOperationName, driverName, snapshot)
+	ctrl.metricsManager.OperationStart(deleteOperation)
+	snapshotStatusType := metrics.SnapshotStatusTypeControllerError
+	defer func() {
+		if err != nil {
```

Comment:
That's true. I'll add a `metrics.SnapshotStatusTypeInProgress`. This will be the default status, and we can update to ControllerError/etc as needed.

This way, we'll always record a metric, but won't have ControllerError by default.